hypershift-operator/metrics.go

```diff
@@ -319,11 +323,15 @@ func (m *hypershiftMetrics) observeHostedClusters(hostedClusters *hyperv1.Hosted
 			if !known {
 				continue
 			}
-			if expectedState {
+			if expectedState == true {
+				// We init to 0, this is needed so the condition is counted down if it goes from the non expectedState to the expectedState.
+				hcByConditions.Init(platform, "not_"+cond.Type)
```

Comment:
@enxebre I don't think this does what we need it to do. This would reset the hc counter to 0 for every hosted cluster. What we expect to have at the end of the loop through all hc's is some tally by error condition of all clusters. For example:

```
not_ValidConfiguration 1
not_Available 2
not_KubeAPIServerAvailable 2
... etc
```

The problem is that because we only count the conditions present in a given hostedcluster, we don't count when that condition is not present.

I think we need to have a list of all possible conditions, then for every hosted cluster in the system iterate through all possible conditions. If that condition is present in the particular hosted cluster, then increase its count by 1. At the end we have a tally of the counts of all hosted clusters for every one of those conditions and that's what we report.

Reply:
I changed the approach to match what we do with idps and removed the platform label for now which makes the logic more complicated and I don't think it's adding much value at least for now, let me know what you think.
I also changed to init the hcCount because right now if all  HC were deleted I don't see how we would report that, I think we would just see the latest value where there was any HC.